sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -146,6 +161,15 @@ def load(*args, **kwargs) -> "AzureAppConfigurationProvider":
     if (endpoint or credential) and connection_string:
         raise ValueError("Please pass either endpoint and credential, or a connection string.")
 
+    # Removing use of AzureAppConfigurationKeyVaultOptions
+    if key_vault_options:
```

Comment:
Generally speaking, `keyvault_credential` in kwarg is more explicit than `key_vault_options`.`credential` hence it should get precedence. (Same for the other two)

Any reason why we don't do it?

Reply:
Backwards compatibility. We have only `AzureAppConfigurationKeyVaultOptions` currently in the GA version. We are soft removing it, but don't want to break what is. currently using it.

Reply:
Wait a minute, is it allowed that user specifies both `key_vault_options` and `keyvault_credential`/`secret_resolver`/`keyvault_client_configs`?

Reply:
I wasn't going to stop them. But only one is needed.

Reply:
What if user provides `keyvault_credential` & `key_vault_options`.`secret_resolver`?

Reply:
Same thing the happens now. Credential wins

Reply:
It looks to me in this case, we will have both `keyvault_credential` & `secret_resolver`?

Shouldn't we check `key_vault_options` only if none of `keyvault_credential`, `secret_resolver`,`keyvault_client_configs` is provided?

Reply:
Wouldn't that be considered a possible breaking change? Those kwargs shouldn't be set but they could be.

Reply:
`keyvault_credential`, `secret_resolver`, `keyvault_client_configs` were not supported in old code, right?

If users explicitly set new kwargs, they opt into the new behavior. :)

Reply:
@xiangyan99 This should be good now. Added the checks and validation.